File: 5주차 과제/시각/Time.py
# 24 * 60 * 60 = 86,400
# 단순히 시각을 1씩 증가시키며 3이 포함되었는지 확인
# 완전탐색 문제

n = int(input())
count = 0

for i in range(n + 1): # 시
    for j in range(60): # 분
        for k in range(60): # 초
            if '3' in str(i) + str(j) + str(k):
                count += 1

print(count)

File: 5주차 과제/시각/Time.c
#include <stdio.h>

int check(int i, int j, int k);

int main(void){
  int n;
  printf("입력할 정수 N : ");
  scanf("%d", &n);
  int count = 0;

  for (int i = 0; i <= n; i++){
    for (int j = 0; j < 60; j++) {
      for (int k = 0; k < 60; k++) {
        if (check(i, j, k)) count++;
      }
    }
  }

  printf("%d", count);
  return 0;
}

int check(int i, int j, int k){
  if (i % 10 == 3 || j / 10 == 3 || j % 10 == 3 || k / 10 == 3 || k % 10 == 3) return 1;
  return 0; // False
}